start.bat

```diff
@@ -1,54 +1,85 @@
 rem Algorithm courtesy of https://github.com/AdoptOpenJDK/IcedTea-Web/blob/master/launchers/shell-launcher/launchers.bat.in
 set "JDK_LOCATION="
+
+rem If the env var JAVA_HOME is not empty, check it points to a functioning executable
 if not "%JAVA_HOME%" == "" (
   "%JAVA_HOME%/bin/java.exe" -version >nul 2>&1
   if errorlevel 0 if not errorlevel 1 (
     set "JDK_LOCATION=%JAVA_HOME%"
   ) else (
     echo Your JDK at [%JAVA_HOME%] read from the JAVA_HOME environment variable is not valid. Please fix this.
+
+    rem Terminate script
     goto:eof
   )
 ) else (
   if "%VERBOSE%" == "TRUE" ( echo JAVA_HOME environment variable non-existent. )
 )
 
+rem Attempt to find java.exe from registry keys
 if "%JDK_LOCATION%" == "" (
+  rem First, find the version
   for /f "tokens=*" %%a in ('%windir%\System32\reg query "HKLM\SOFTWARE\JavaSoft\Java Development Kit" 2^>nul') do set "_version_key=%%a"
+
+  rem Remove the first 57 chars, the length of "HKEY_LOCAL_MACHINE\SOFTWARE\JavaSoft\Java Development Kit"
   set "_version=!_version_key:~58!"
+
+  rem Second, query for a specific registry key value of JavaHome
   for /f "tokens=*" %%a in ('%windir%\System32\reg query "HKLM\SOFTWARE\JavaSoft\Java Development Kit\!_version!" /v JavaHome 2^>nul') do set "_jh_key=%%a"
+
+  rem Remove the first 21 chars, the length of ???
   set "_BAD_SLASH_JAVA_HOME=!_jh_key:~22!"
+
+  rem Replace backslashes with forward slashes
   set "_REG_JAVA_HOME=!_BAD_SLASH_JAVA_HOME:\=/!"
 
+  rem Check that the found java.exe path points to a functioning executable
   if EXIST "!_REG_JAVA_HOME!/bin/java.exe" (
+    rem Test exit_success of calling "java -version, ignoring outputs"
     "!_REG_JAVA_HOME!/bin/java.exe" -version >nul 2>&1
-      if errorlevel 0 if not errorlevel 1 (
-        set JDK_LOCATION=!_REG_JAVA_HOME!
-      ) else (
-        echo Your JDK [!_REG_JAVA_HOME!] read from Registry HKLM\SOFTWARE\JavaSoft\Java Development Kit is not valid. Please fix this.
-        goto:eof
-      )
+    if errorlevel 0 if not errorlevel 1 (
+      set JDK_LOCATION=!_REG_JAVA_HOME!
+    ) else (
+      echo Your JDK [!_REG_JAVA_HOME!] read from Registry HKLM\SOFTWARE\JavaSoft\Java Development Kit is not valid. Please fix this.
+
+      rem Terminate script
+      goto:eof
+    )
   ) else (
     if "%VERBOSE%" == "TRUE" ( echo Registry entries for JAVA_HOME non-existent. )
   )
+
+  rem Unset variables for some reason
   set "_version_key=" & set "_version=" & set "_jh_key=" & set "_BAD_SLASH_JAVA_HOME=" & set "_REG_JAVA_HOME="
 )
 
+rem Attempt to find java.exe from the PATH environment variable
 if "%JDK_LOCATION%" == "" (
   set "_PATH_JAVA_HOME="
+
+  rem Split PATH on ";" (Windows' path separator), iterate over the chunks
   for %%a in ("%PATH:;=";"%") do (
     if exist "%%~a/javac.exe" (
+      rem Somehow get parent directory (from ."/bin/" to "./")? Might be a typo/oversight with both calls getting the drive+path of %%a, might not
```

Comment:
right. This seems to be broken now. As it NEVER tries the correct executable.
Currently, it finds javac.exe on path, and then it strips it, so it never works. Also it strips to much, but it was uncelar froim outputs.